programs/server/Server.cpp

```diff
@@ -1486,16 +1486,21 @@ try
 
     /// Load global settings from default_profile and system_profile.
     global_context->setDefaultProfiles(config());
-    const Settings & settings = global_context->getSettingsRef();
 
     /// Initialize background executors after we load default_profile config.
     /// This is needed to load proper values of background_pool_size etc.
     global_context->initializeBackgroundExecutorsIfNeeded();
 
-    if (settings.async_insert_threads)
+    size_t async_insert_threads = config().getUInt("async_insert_threads", 16);
+    bool async_insert_queue_flush_on_shutdown = config().getBool("async_insert_queue_flush_on_shutdown", false);
```

Comment:
Should we turn it on by default?

Reply:
I think we should try to flash it when gracefull shutdown.